utils/mgmt_system/openstack.py

```diff
@@ -201,7 +201,16 @@ def create_vm(self):
     def delete_vm(self, instance_name):
         logger.info(" Deleting OpenStack instance %s" % instance_name)
         instance = self._find_instance_by_name(instance_name)
+        ip_addr = self.current_ip_address(instance_name)
+        floating_ip = None
+        if ip_addr is not None:
+            floating_ips = self.api.floating_ips.findall(ip=ip_addr)
+            if floating_ips:
+                floating_ip = floating_ips[0]
         instance.delete()
+        if floating_ip is not None:
+            logger.info("Deleting floating IP {}".format(floating_ip.ip))
```

Comment:
Do we need to delete the FIP? Can't it just be reassigned? Giving us a lesser chance that the pool will be depleted before we get it due to the nova/neutron bug?